### PR TITLE
Change the current architecture to keep alive architecture

### PR DESCRIPTION
**Pull request by AakashMallik**
Fixes #247  #242

## Changes
- [x] Converted the python and electron communication architecture to **keep alive**
- [x] Tested it for 10k samples, for 10usec tg and **30** milsec delay
- [x] Testing
- [x] Electron-load-balancer lib upgraded to 1.0.6
- [x] Peer testing

### CODE REVIEW

**Comment by AakashMallik** [comment]
@fragm3 @mariobehling @CloudyPadmal Don't merge this as of now. Let me get everyone to test this first. This architrecture will help us in the long run. Also, the app holds well at 30milisec delay to which even I was suprised. :laughing:

@fragm3 @CloudyPadmal Please test this branch on your machines as you two have PSLab kits with you. Make sure to do a ```npm install``` before u start as I had to update my library to compensate for the changes.

File: src/App.js [hunk 1]
```diff
@@ -8,12 +8,29 @@ import LogicAnalyser from './screen/LogicAnalyser';
...
 import './App.css';
-
 const electron = window.require('electron');
-const fs = electron.remote.require('fs');
 const ipcRenderer = electron.ipcRenderer;
+const loadBalancer = window.require('electron-load-balancer');
 
 class App extends Component {
+	componentDidMount() {
+		loadBalancer.startBackgroundProcess(ipcRenderer, 'oscilloscope', {
```
Comment (fragm3) [inline src/App.js]: We have to handle this for many instruments

Comment (fragm3) [inline src/App.js]: We have to handle this for many instruments

File: src/App.js [hunk 2]
```diff
@@ -8,12 +8,29 @@ import LogicAnalyser from './screen/LogicAnalyser';
...
+				ch2: false,
+				ch3: false,
+				ch4: false,
+			},
+			delay: 100,
+		});
+	}
+
+	componentWillUnmount() {
+		loadBalancer.stopBackgroundProcess(ipcRenderer, 'oscilloscope');
```
Comment (fragm3) [inline src/App.js]: Same here

File: src/App.js (hunk 1 above)
Comment (AakashMallik) [reply src/App.js]: Naming issue from my side. We will just have to deal with a single background process now. I'll change this accordingly..

File: src/App.js (hunk 2 above)
Comment (AakashMallik) [reply src/App.js]: Same
8

File: src/App.js (hunk 1 above)
Comment (AakashMallik) [reply src/App.js]: @fragm3 I have addressed this issue in my latest PR https://github.com/fossasia/pslab-desktop/pull/249
I have renamed the background process as ```linker``` and hopefully we'll stick to having just one background process like we discussed today.

**Comment by fragm3** [comment]
@AakashMallik I have tested it, the implementation works fine.

**Pull request closed without merging**